src/structures/Role.js

```diff
@@ -406,10 +406,8 @@ class Role extends Base {
    *   .then(deleted => console.log(`Deleted role ${deleted.name}`))
    *   .catch(console.error);
    */
-  async delete(reason) {
-    await this.client.api.guilds[this.guild.id].roles[this.id].delete({ reason });
-    this.client.actions.GuildRoleDelete.handle({ guild_id: this.guild.id, role_id: this.id });
-    return this;
+  delete(reason) {
+    return this.guild.roles.delete(this.id, reason);
```

Comment:
Alongside the other suggestion:
```suggestion
  async delete(reason) {
    await this.guild.roles.delete(this.id, reason);
    return this;
```